### DISCUSSION TITLE
How to connect to NASA protocol with wire F3 F4

### DISCUSSION THREAD

**Discussion by ngtrangtrung** [omerfaruk-aran/esphome_samsung_hvac_bus#56] [Questions]
Hello 
Wire Remote Controller, With Nasa there are only 2 lines F3 F4, how will the connection line be connected?
![1280X1280](https://github.com/lanwin/esphome_samsung_ac/assets/86597628/ec93db3d-de2d-4f02-96a6-714c80da0324)
Can I use M5STACK ATOM to connect like non-nasa?

**Comment by lanwin** [comment]
@ngtrangtrung I am not sure what we see on your picture. This project is designed to be used for the connection between the outdoor unit and the indoor unit. Not for the connection between remote controller and indoor unit.

**Comment by atanasenko** [comment]
So I've gotten a hold of MIM-A00N sub-pcb to provide F3/F4 connection and a MWR-WG00JN wired controller for one of the indoor units. Quick test shows that F1/F2 only supplies minimal signal voltage, but F3/F4 is actually used to power the wired remote along with passing data around and there's around 10-12V.
My idea is to listen for what the controller sends to indoor unit to find something useful, but I'm afraid to connect my m5 atom to that bus. @lanwin, do you know if this "power over rs-485" feature a standard one and is usually handled correctly?

![20240521_124316](https://github.com/lanwin/esphome_samsung_ac/assets/266280/99eb1ff2-01da-418e-83ad-7a7f275b3dba)
![20240521_124338](https://github.com/lanwin/esphome_samsung_ac/assets/266280/32cca7a4-e19f-4764-af61-f1020f2aea95)
![20240521_124348](https://github.com/lanwin/esphome_samsung_ac/assets/266280/0fa38288-b250-4873-b8db-7317e663b433)

**Reply by lanwin** [reply]
Sorry but I have no clue. I am not a hardware guy. Maybe some of the other users can help.

**Comment by mchiriciuc** [comment]
You can decouple the atom form DC on the F3/F4 bus by connecting the A and B lines to F3 and F4 with 10uF/25V ceramic capacitors.
I want to do the same, but by connecting the Atom to RX and TX on the 14 pin connector. 
I think a better way of getting rid of smartthings is to replace the MIM-A00N sub-pcb with a custom pcb with an esp directly connected to the indoor unit's UART and emulate MWR-WG00JN wired controller.
Can you turn internal units BEEP and display on/off them the wired controller?
Or to replace the wireless module with a custom one. (still UART)

**Reply by stresserini** [reply]
Hi mchiriciuc
Were you able to get the data reading to work with F3 and F4?
I would like to try this idea, but I have a question to ask.

If I can read the data via F3/F4, how do I program the ESP with the relevant entities?
I imagine they are not the same as the data coming from F1/F2
Mark

**Reply by mchiriciuc** [reply]
Hi stresserini,
I was not able to get my hands on the required sub-pcb and a wired controller yet.
I would connect the ESP si the same program to F3/F4 since it is also NASA protocol as far as I know, just to see and compare.
